pkg/dao/types/property/field.go

```diff
@@ -425,29 +425,24 @@ func (i Property) Cty() (cty.Type, cty.Value, error) {
 	}
 
 	switch {
-	// From Terraform expression guide,
-	// https://developer.hashicorp.com/terraform/language/expressions/type-constraints#collection-types,
-	// map(any), list(any), set(any) are the compatible expansions for map, list, set.
-	case (i.Type.IsMapType() && cty.DynamicPseudoType.Equals(*i.Type.MapElementType())) ||
-		(i.Type.IsListType() && cty.DynamicPseudoType.Equals(*i.Type.ListElementType())) ||
-		(i.Type.IsSetType() && cty.DynamicPseudoType.Equals(*i.Type.SetElementType())):
-		// When the property type is very loose,
-		// only perform conversion checks on its value.
+	case i.Type == cty.NilType || i.Type == cty.DynamicPseudoType:
```

Comment:
Can we add unit tests to ensure it works in different cases?

Reply:
updated